src/store.rs

```diff
@@ -68,3 +69,52 @@ impl<A: Hash + Eq + Display, B: Display + Clone> Store<A, B> {
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    #[test]
+    fn test_get_fails_if_the_storage_not_does_contain_the_key() {
+        let store: Store<&str, &str> = Store::new();
+
+        let result = store.get("key1");
+        assert_eq!(result, Err(ExecResult::Failed));
+    }
+
+    #[test]
+    fn test_set_fails_if_the_storage_already_contains_the_key() {
+        let mut store = Store::new();
+
+        let result = store.set("key1", "value1");
+        assert_eq!(result, ExecResult::Success);
+
+        let result = store.set("key1", "value1");
+        assert_eq!(result, ExecResult::Failed);
+    }
+
+    #[test]
+    fn test_del_fails_if_the_storage_not_does_contain_the_key() {
```

Comment:
rename `test_del_fails_if_the_storage_not_does_contain_the_key()` as `test_del_key_not_found()`

Reply:
Also, is it a good idea to use a singular test for both `del` and `get`?

In that case we can call it `test_key_not_found()`. So we will be testing the various components on the basis of the circumstance, instead of their feature, so less redundancy, I might be wrong here. 

We can continue this discussion in #6 to provide more concrete expectations from the testing strategy.